src/Hit/Git.hs

```diff
@@ -260,21 +261,47 @@ issueFromBranch =
     . T.drop 1
     . T.dropWhile (/= '/')
 
+
+{- | Perform the given action by first staging the given files and
+later removing them again after the action
+-}
+withFiles :: IO a -> IO [Text] -> ([Text] -> IO ()) -> IO a
+withFiles action whichFiles removeFiles = bracket
+    addFiles
+    removeFiles
+    (const action)
+  where
+    addFiles :: IO [Text]
+    addFiles = do
+        files <- whichFiles
+        for_ files $ \file -> void $ "git" $| ["add", file]
+        pure files
+
+{- | Perform given action by adding all deleted files to index and returning
+them back after action.
+-}
+withDeletedFiles :: IO a -> IO a
+withDeletedFiles action = withFiles action deletedFiles removeDeletedFiles
+  where
+    -- Find the deleted file to index so they will appear in diff
+    deletedFiles :: IO [Text]
+    deletedFiles = lines <$> "git" $| ["ls-files", "-d", "--exclude-standard"]
```

Comment:
Let's write in a more verbose way to make code easier to understand :slightly_smiling_face: Since these commands are automated, code readability and maintainability is more important 
```suggestion
    deletedFiles = lines <$> "git" $| ["ls-files", "--deleted", "--exclude-standard"]
```